tests/metrics/classification/test_inputs.py

```diff
@@ -116,42 +116,41 @@ def _mlmd_prob_to_mc_preds_tr(x):
     [
         #############################
         # Test usual expected cases
-        (_bin, None, False, None, "multi-class", _usq, _usq),
-        (_bin, 1, False, None, "multi-class", _usq, _usq),
-        (_bin_prob, None, None, None, "binary", lambda x: _usq(_thrs(x)), _usq),
-        (_ml_prob, None, None, None, "multi-label", _thrs, _idn),
-        (_ml, None, False, None, "multi-dim multi-class", _idn, _idn),
-        (_ml_prob, None, None, None, "multi-label", _ml_preds_tr, _rshp1),
-        (_ml_prob, None, None, 2, "multi-label", _top2, _rshp1),
-        (_mlmd, None, False, None, "multi-dim multi-class", _rshp1, _rshp1),
-        (_mc, NUM_CLASSES, None, None, "multi-class", _onehot, _onehot),
-        (_mc_prob, None, None, None, "multi-class", _top1, _onehot),
-        (_mc_prob, None, None, 2, "multi-class", _top2, _onehot),
-        (_mdmc, NUM_CLASSES, None, None, "multi-dim multi-class", _onehot, _onehot),
-        (_mdmc_prob, None, None, None, "multi-dim multi-class", _top1_rshp2, _onehot),
-        (_mdmc_prob, None, None, 2, "multi-dim multi-class", _top2_rshp2, _onehot),
-        (_mdmc_prob_many_dims, None, None, None, "multi-dim multi-class", _top1_rshp2, _onehot_rshp1),
-        (_mdmc_prob_many_dims, None, None, 2, "multi-dim multi-class", _top2_rshp2, _onehot_rshp1),
+        (_bin, None, False, None, DataType.MULTICLASS, _usq, _usq),
+        (_bin, 1, False, None, DataType.MULTICLASS, _usq, _usq),
+        (_bin_prob, None, None, None, DataType.BINARY, lambda x: _usq(_thrs(x)), _usq),
+        (_ml_prob, None, None, None, DataType.MULTILABEL, _thrs, _idn),
+        (_ml, None, False, None, DataType.MULTIDIM_MULTICLASS, _idn, _idn),
+        (_ml_prob, None, None, None, DataType.MULTILABEL, _ml_preds_tr, _rshp1),
+        (_mlmd, None, False, None, DataType.MULTIDIM_MULTICLASS, _rshp1, _rshp1),
+        (_mc, NUM_CLASSES, None, None, DataType.MULTICLASS, _onehot, _onehot),
+        (_mc_prob, None, None, None, DataType.MULTICLASS, _top1, _onehot),
+        (_mc_prob, None, None, 2, DataType.MULTICLASS, _top2, _onehot),
+        (_mdmc, NUM_CLASSES, None, None, DataType.MULTIDIM_MULTICLASS, _onehot, _onehot),
+        (_mdmc_prob, None, None, None, DataType.MULTIDIM_MULTICLASS, _top1_rshp2, _onehot),
+        (_mdmc_prob, None, None, 2, DataType.MULTIDIM_MULTICLASS, _top2_rshp2, _onehot),
+        (_mdmc_prob_many_dims, None, None, None, DataType.MULTIDIM_MULTICLASS, _top1_rshp2, _onehot_rshp1),
+        (_mdmc_prob_many_dims, None, None, 2, DataType.MULTIDIM_MULTICLASS, _top2_rshp2, _onehot_rshp1),
```

Comment:
lest passing it as a string...
also thinking that we can have inside this func a loop when we test string/enum
```
for exp_mode in (exp_mode, str(exp_mode)):
    ...
```

Reply:
found that `str(DataType.BINARY) == 'DataType.BINARY'` 

figured one way to have the desired effect( `str(DataType.BINARY) == 'binary'`) is to add
```
def __str__(self):
    return self.value.lower()
```
to `LightningEnum`

Reply:
and think better with
```
for exp_mode in (exp_mode, DataType.from_str(exp_mode)):
 ...
```
while keeping input param str.

Reply:
> figured one way to have the desired effect( `str(DataType.BINARY) == 'binary'`) is to add

why would you need it since you can compare LEnum with string?
see https://github.com/PyTorchLightning/pytorch-lightning/blob/a17c941cc4ec459514f5a465f5bf3af15351b54e/pytorch_lightning/utilities/enums.py#L44

Reply:
thought you meant
```
for exp_mode in (exp_mode: Enum, str(exp_mode: Enum)):
...
```
when `exp_mode` is `DataTaype.Binary` then `str(exp_mode) == "DataTaype.Binary"`

Reply:
my main point was to test both inputs as string and enum and the simples way would be run the very same test just once enter it as string and later as enum, that is the reason for the for loop